File: spec/lua/extra/my_custom_fx.lua
local class = require("class")

MyCustomFx = class.class()

function MyCustomFx:_init(key, flag, some_bytes)
    if flag then
        self._key = key
    else
        self._key = -key
    end
end

function MyCustomFx:decode(data)
    local r = ""

    for i = 1, #data do
        local c = (data:byte(i) + self._key) % 0x100
        r = r .. string.char(c)
    end

    return r
end

File: spec/python/extra/my_custom_fx.py
class MyCustomFx:
    def __init__(self, key, flag, some_bytes):
        self.key = key if flag else -key

    def decode(self, data):
        r = bytearray(data)
        for i, byte in enumerate(r):
            r[i] = (byte + self.key) % 0x100
        return bytes(r)

    def encode(self, data):
        old_key = self.key
        self.key = -self.key
        res = self.decode(data)
        self.key = old_key
        return res
